### DISCUSSION TITLE
A beginner has a problem in a real project and asks for a solution.

### DISCUSSION THREAD

**Discussion by LingClassmate** [argoproj/argo-events#1749] [General]
My workflow:
start——>Task1——>Task2——>Task3——>end

My application scenario:  
After Workflow Task1 is executed, wait for Events to trigger task2. After Task2 is executed, you do not need to trigger task3.

Thank you very much!

**Comment by LingClassmate** [comment]
I want to be able to execute the Workflow task first, and then set an events in the workflow to trigger the subsequent task flow.

**Comment by juliev0** [comment]
Sorry for a delayed response. Can you clarify what you mean? You should be able to have a single set of Steps in a Workflow without needing Argo Events in between them. Argo Events would typically be used to start up a new Workflow (set of steps), but you could also watch for the termination of your Workflow with Argo Events to start up a new Workflow. To do that, you could either use a Resource Type Event Source that watches for the completion of your Workflow, something like this:
```
#    # watch for completed workflows in any namespace
#    example-without-namespace:
#      # namespace: (omitted to match any namespace)
#      group: "k8s.io"
#      version: v1
#      resource: workflows
#      eventTypes:
#        - ADD
#      filter:
#        prefix: "my-workflow"
#        labels:
#          - key: workflows.argoproj.io/completed
#            operation: ==
#            value: "true"
```
or you could add an extra step in the first workflow to send a message to a queue or pub/sub), and use argo events to watch this event.
